internal/sgbouncewizard/sgbouncewizard.go

```diff
@@ -577,6 +570,50 @@ func (srv *Server) Serve(Port int) {
 		})
 	})
 
+	r.Route("/ws", func(r chi.Router) {
+		r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+			m.HandleRequest(w, r)
+		})
+	})
+
+	m.HandleMessage(func(s *melody.Session, msg []byte) {
+		log.Println(string(msg))
+		params := strings.Split(string(msg), " ")
+		lock.Lock()
+		if params[0] == "edit" {
+			if currentlyViewing[params[1]] {
+				_, exists := s.Get(params[1])
+				if !exists {
+					log.Println("ATTEMPT TO EDIT RULE THAT IS BEING EDITED")
+					s.Write([]byte("INUSE"))
+				} else {
+					s.Write([]byte("ALREADY"))
+				}
+			} else {
+				s.Set(params[1], true)
+				currentlyViewing[params[1]] = true
+				s.Write([]byte("EDIT"))
+				m.BroadcastOthers([]byte("INUSE"), s)
```

Comment:
What is the purpose of these `BroadcastOthers`? It looks like they'll just receive a message that says `INUSE` or `FREE`, which doesn't seem very useful?

Reply:
I believe my idea of the socket broadcasting `INUSE` to everyone but the current connection is to be able to switch their currently frontend UI to a different view if they were both currently on the same page